lib/response.js

```diff
@@ -229,7 +242,39 @@ res.send = function send(body) {
     this.end();
   } else {
     // respond
-    this.end(chunk, encoding);
+    if (isChunkBlob) {
+      /**
+       * @type {http.ServerResponse}
+       */
+      var res = this;
+      /**
+       * @type {Blob}
+       */
+      var blob = chunk;
+      var destroyWriteStream = function (err) {
+        if (res.writable) {
+          next(err);
+        }
+      }
+      if(typeof blob.stream === 'function') {
+        var inputStream = Readable.from(blob.stream());
+        inputStream.on('error', destroyWriteStream);
+        res.on('error',function(err){
+          destroyWriteStream(err)
+          inputStream.unpipe(res);
```

Comment:
Is there a reason for this order? Shouldn't the stream be `unpipe`d first and then the error propagated, or would `unpipe` cause the stream to become un`writable`?

Reply:
ok, just change the line of execution.